src/recipes.jl

```diff
@@ -801,6 +801,50 @@ end
 # note: don't add dependencies because this really isn't a drop-in replacement
 
 
+# ---------------------------------------------------------------------------
+# lens! - magnify a region of a plot
+@recipe function f(::Type{Val{:lens}}, plt::AbstractPlot)
```

Comment:
I'm surprised this works - it's the type recipe signature, which is usually just used for type conversions to vector. Should this not be a "user recipe" with a `@userplot` shorthand defined?

Reply:
I thought, this is the signature of a [plot recipe](https://docs.juliaplots.org/latest/recipes/#plot-recipes)

Reply:
also I would like to only define the `lens!` shorthand, since `lens` by itself doesn't make too much sense.

Reply:
Ah you're right. I don't think they are used anywhere in the ecosystem, but I may be wrong (it lists MarginalHist as an example, but looking down on that page clearly shows that marginalhist is a "user recipe"). Defining a `lens!` shorthand makes sense - but I feel that calling it with `seriestype` as done here feels unideomatic.

Reply:
What would be a more ideomatic alternative? I don't bother too much to change this.

Reply:
Just the lens shorthand? Sorry I have zero experience using plot recipes

Reply:
Okay, now I got it. You meant the way to call the recipe is unideomatic, I thought the implementation with using the `seriestype`. I will add the `lens!` shorthand.